RocketChat/Rocket.Chat
# [IMPROVE] Add user to room on "Click to Join!" button press

## Proposed changes (including videos or screenshots)


- Add user to room on "Click to Join!" button press;
- Display the "Join" button in discussions inside channels (keeping the behavior consistent with discussions inside groups).



## Issue(s)

Fix #19420

## Steps to test or reproduce

1. Start a new Jitsi call in a channel or discussion (logged in as a member of the room). Then, access the room as another user and press the "Click to Join!" button. **_Expected behavior:_** the user should join the room (if they're not already a member).
2. Access a discussion inside a **channel** (public room). **_Expected behavior:_** If the user is not a member of the discussion, a "Join" button should be displayed in the message box.

## Further comments
